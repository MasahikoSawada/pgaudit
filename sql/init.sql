--
-- Note:
-- In order to install pgaudit first, the initial pgaudit config
-- file is 'empty.conf' located in conf/ directory. After installed
-- pgaudit, change config file to properly file 'audit.conf' and
-- restart it.
--
CREATE EXTENSION pgaudit;
-- check installation.
SELECT extname, extversion FROM pg_extension;
-- check pgaudit event triggers.
SELECT evtname, evtevent FROM pg_event_trigger WHERE evtname LIKE 'pgaudit%';

-- Install pgaudit in other databases
SELECT current_database() \gset
CREATE DATABASE foo;
CREATE DATABASE bar;
\c foo
CREATE EXTENSION pgaudit;
SELECT extname, extversion FROM pg_extension;
SELECT evtname, evtevent FROM pg_event_trigger WHERE evtname LIKE 'pgaudit%';
\c bar
CREATE EXTENSION pgaudit;
SELECT extname, extversion FROM pg_extension;
SELECT evtname, evtevent FROM pg_event_trigger WHERE evtname LIKE 'pgaudit%';

-- Change config file to properly file.
ALTER SYSTEM SET pgaudit.config_file TO '../../conf/audit.conf';
\! pg_ctl restart -w -D ./tmp_check/data > /dev/null
-- Connection will be disconnected, close connection explicitly .
\q
