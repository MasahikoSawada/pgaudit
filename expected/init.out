--
-- Note:
-- In order to install pgaudit first, the initial pgaudit config
-- file is 'empty.conf' located in conf/ directory. After installed
-- pgaudit, change config file to properly file 'audit.conf' and
-- restart it.
--
CREATE EXTENSION pgaudit;
-- check installation.
SELECT extname, extversion FROM pg_extension;
 extname | extversion 
---------+------------
 plpgsql | 1.0
 pgaudit | 1.0
(2 rows)

-- check pgaudit event triggers.
SELECT evtname, evtevent FROM pg_event_trigger WHERE evtname LIKE 'pgaudit%';
         evtname         |    evtevent     
-------------------------+-----------------
 pgaudit_ddl_command_end | ddl_command_end
 pgaudit_sql_drop        | sql_drop
(2 rows)

-- Install pgaudit in other databases
SELECT current_database() \gset
CREATE DATABASE foo;
CREATE DATABASE bar;
\c foo
CREATE EXTENSION pgaudit;
SELECT extname, extversion FROM pg_extension;
 extname | extversion 
---------+------------
 plpgsql | 1.0
 pgaudit | 1.0
(2 rows)

SELECT evtname, evtevent FROM pg_event_trigger WHERE evtname LIKE 'pgaudit%';
         evtname         |    evtevent     
-------------------------+-----------------
 pgaudit_ddl_command_end | ddl_command_end
 pgaudit_sql_drop        | sql_drop
(2 rows)

\c bar
CREATE EXTENSION pgaudit;
SELECT extname, extversion FROM pg_extension;
 extname | extversion 
---------+------------
 plpgsql | 1.0
 pgaudit | 1.0
(2 rows)

SELECT evtname, evtevent FROM pg_event_trigger WHERE evtname LIKE 'pgaudit%';
         evtname         |    evtevent     
-------------------------+-----------------
 pgaudit_ddl_command_end | ddl_command_end
 pgaudit_sql_drop        | sql_drop
(2 rows)

-- Change config file to properly file.
ALTER SYSTEM SET pgaudit.config_file TO '../../conf/audit.conf';
\! pg_ctl restart -w -D ./tmp_check/data > /dev/null
-- Connection will be disconnected, close connection explicitly .
\q
